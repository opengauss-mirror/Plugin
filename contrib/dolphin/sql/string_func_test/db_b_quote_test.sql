drop database if exists test;
create database test dbcompatibility 'B';
\c test

SELECT QUOTE(E'Don\'t!');
SELECT QUOTE(E'O\'hello');
SELECT QUOTE('O\hello');

\c postgres
drop database test;