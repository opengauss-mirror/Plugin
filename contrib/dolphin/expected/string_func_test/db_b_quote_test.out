create schema db_b_quote_test;
set current_schema to 'db_b_quote_test';
SELECT QUOTE(E'Don\'t!');
   quote   
-----------
 'Don''t!'
(1 row)

SELECT QUOTE(E'O\'hello');
   quote    
------------
 'O''hello'
(1 row)

SELECT QUOTE('O\hello');
    quote    
-------------
 E'O\\hello'
(1 row)

SELECT QUOTE(char(89));
 quote 
-------
 'Y'
(1 row)

SELECT QUOTE(char(224));
ERROR:  Cannot convert string 0xe0 to UTF8
CONTEXT:  referenced column: quote_literal
referenced column: quote_literal
SQL statement "SELECT (select quote_literal(t))"
PL/pgSQL function quote(text) line 3 at RETURN
referenced column: quote
drop schema db_b_quote_test cascade;
reset current_schema;
